File: OpenLabHub/OpenLabHub/OpenLabHub/Models/Project.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace OpenLabHub.Models
{
    public class Project
    {
        public string title { get; set; }
        public string description { get; set; }
        private DateTime dateCreated { get; set; }
        public string state { get; set; }
        public string imageUrl { get; set; }
        public HashSet<ProjectTag> projectTags { get; set; }
        public User user { get; set; }
        

    }
    public class ProjectTag
    {
        public Project project { get; set; }
        public float weight { get; set; }
        public Tag tag { get; set; }
    }
}

File: API/SpaceAppChallenge/src/main/java/com/example/spaceappchallenge/model/Project.java
package com.example.spaceappchallenge.model;

import com.example.spaceappchallenge.enums.State;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name="project")
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long projectId;
    private String title;
    private String description;
    private LocalDate dateCreated;
    private String status;
    private String imageUrl;
    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(
            name="project_tag",
            joinColumns = @JoinColumn(name="project_id"),
            inverseJoinColumns = @JoinColumn(name="tag_id")
    )
    private Set<Tag> relatedTags;
    @ManyToOne(cascade = CascadeType.PERSIST)
    @JsonIgnore
    @JoinColumn(name="user_id", nullable = false)
    private User user;
}
